pennylane/pauli/grouping/group_observables.py

```diff
@@ -119,75 +152,326 @@ def binary_repr(self, n_qubits=None, wire_map=None):
-                adj = (mat_prod + 1) % 2
-                np.fill_diagonal(adj, 0)
+    def _partition_custom_indices(self, observables_indices) -> list[list]:
+        """Compute the indices of the partititions of the observables when these have custom indices.
 
-        return adj
+        TODO: Use this function to calculate custom indices instead of calculating observables first.
 
-    def colour_pauli_graph(self):
+        Args:
+            observables_indices (tensor_like, optional): A tensor or list of indices associated to each observable.
+                This argument is helpful when the observables used in the graph colouring are part of a bigger set of observables.
+                Defaults to None.
+
+        Returns:
+            tuple[tuple[int]]: Tuple of tuples containing the indices of the observables on each partition.
         """
-        Runs the graph colouring heuristic algorithm to obtain the partitioned Pauli words.
+        partition_indices = items_partitions_from_idx_partitions(
+            observables_indices, self._idx_partitions_dict_from_graph.values(), return_tuples=True
+        )
+
+        return partition_indices
+
+    def pauli_partitions_from_graph(self) -> list[list]:
+        """Partition Pauli observables into lists of (anti-)commuting observables
+        using ``Rustworkx`` graph colouring algorithms based on binary relation determined by  ``self.grouping_type``.
 
         Returns:
-            list[list[Observable]]: a list of the obtained groupings. Each grouping is itself a
-            list of Pauli word ``Observable`` instances
+            list[list[Observable]]: List of partitions of the Pauli observables made up of mutually (anti-)commuting terms.
         """
+        # Get the observables from the indices. itemgetter outperforms list comprehension
+        pauli_partitions = items_partitions_from_idx_partitions(
+            self.observables, self._idx_partitions_dict_from_graph.values()
+        )
+        return pauli_partitions
 
-        if self.adj_matrix is None:
-            self.adj_matrix = self.complement_adj_matrix_for_operator()
 
-        coloured_binary_paulis = self.graph_colourer(self.binary_observables, self.adj_matrix)
+def items_partitions_from_idx_partitions(
+    items: Sequence, idx_partitions: Sequence[Sequence[int]], return_tuples: bool = False
+) -> Sequence[Sequence]:
+    """Get the partitions of the items corresponding to the partitions of the indices.
 
-        self.grouped_paulis = [
-            [binary_to_pauli(pauli_word, wire_map=self._wire_map) for pauli_word in grouping]
-            for grouping in coloured_binary_paulis.values()
+    Args:
+        items (Sequence): A Sequence of items to be partitioned according to the partition of the indices.
+        idx_partitions (Sequence[Sequence[int]]): Sequence of sequences containing the indices of the partitioned items.
+        return_tuples (bool): Whether to return tuples of tuples or list of lists.
```

Comment:
Something in particular being sensitive to tuples versus list? I might create a follow up task to standardize everything to one or the other.

Reply:
I agree. The problem is that in applications where the indices were requested, they were somehow expected to be a tuple (they were using some methods that would require them to be tuples)

Reply:
Mind pointing me to a specific case? I'd like to standardize them to one or the other (preferably tuples) if possible.

Reply:
I guess the first one that comes to mind is this line: 

https://github.com/PennyLaneAI/pennylane/blob/794be4b9ea71d5e0c6a0b1d9251fa0023ba99b3f/pennylane/ops/qubit/hamiltonian.py#L464-L465